extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/HttpInstrumenterVertxTracer.java

```diff
@@ -357,6 +358,27 @@ public List<String> responseHeader(final HttpRequest request, final HttpResponse
         }
     }
 
+    private static class ClientSpanNameExtractor implements SpanNameExtractor<HttpRequest> {
+
+        @Override
+        public String extract(HttpRequest httpRequest) {
+            if (httpRequest instanceof HttpRequestHead) {
+                HttpRequestHead head = (HttpRequestHead) httpRequest;
+                if (head.traceOperation != null) {
+                    return head.traceOperation;
+                }
+            }
+            if (httpRequest instanceof WriteHeadersHttpRequest) {
+                WriteHeadersHttpRequest writeHeaders = (WriteHeadersHttpRequest) httpRequest;
+                String traceOperation = writeHeaders.traceOperation();
```

Comment:
There seems to be a case where the Vert.x `traceOperation` is initialized with the `absoluteUri`. Maybe we don't touch that code path, but hard to tell without trying it.

Reply:
Where are you seeing this? `traceOperation` should be null if not explicitly set.

Reply:
Without this code path, most of the trace operations will not be used. All the clients we are using are translated into `WriteHeadersHttpRequest` for reporting.

Reply:
> There seems to be a case where the Vert.x traceOperation is initialized with the absoluteUri.

You might be thinking of the operation name passed to the `VertxTracer`. This operation is never used and not passed on in the `HttpRequest` created for telemetry; that's why I'm interrogating the original `HttpClientRequest`.